app/main.py

```diff
@@ -1,3 +1,79 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    @staticmethod
+    def check_value(value: [Distance, int]) -> int:
+        return (
+            value.real if "real" in dir(value)
+            else value.km
+        )
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: [Distance, int]) -> Distance:
+        return Distance(
+            km=self.km + self.check_value(other)
+        )
+
+    def __iadd__(self, other: [Distance, int]) -> Distance:
+        self.km += self.check_value(other)
+        return self
+
+    def __mul__(self, other: [Distance, int]) -> Distance:
+        if isinstance(other, Distance):
+            raise TypeError("Cannot multiply one Distance instance by another")
+
+        elif isinstance(other, (int, float)):
+            return Distance(
+                km=self.km * other
+            )
+        else:
+            raise TypeError(
+                f"Unsupported type for multiply: {type(other).__name__}"
+            )
+
+    def __truediv__(self, other: [Distance, int]) -> Distance:
+        if isinstance(other, Distance):
+            raise TypeError("Cannot divide one Distance instance by another")
+        elif isinstance(other, (int, float)):
+            return Distance(
+                round((self.km / other), 2)
```

Comment:
The `__truediv__` method should instantiate a `Distance` object with the `km` parameter. Currently, it is missing the `km=` keyword, which might lead to incorrect object creation.